Implemented the QR Code scan functionality.

ayush-sharma2601 opened the pull request:
used https://github.com/yuriy-budiyev/code-scanner, an open source repo based on ZXing.

- created Activity for scanning purpose
- layout file contains codeScannerView
- in Kotlin file, used Easy permission to get camera permissions on creation of activity
- then QR code is scanned automatically and its text is returned in call back

-- added required dependencies(EasyPermissions & CodeScanner) and permissions(Camera)

--layout screen--
![Screenshot (27)](https://user-images.githubusercontent.com/56927212/89043044-86814d80-d365-11ea-9a5f-d563e5ad4f54.png)

ayush-sharma2601 committed:
Implemented the QR Code scan functionality.
used https://github.com/yuriy-budiyev/code-scanner, an open source repo based on ZXing.

- created Activity for scanning purpose
- layout file contains codeScannerView
- in Kotlin file, used Easy permission to get camera permissions on creation of activity
- then qr code is scanned automatically and its text is returned in call back

-- added required dependencies(EasyPermissions & CodeScanner) and permissions(Camera)

devansh-299 commented on app/build.gradle:
```diff
@@ -47,4 +47,10 @@ dependencies {
+    implementation 'com.budiyev.android:code-scanner:2.1.0'
+
+    // For Handling permissions
+    implementation 'pub.devrel:easypermissions:3.0.0'
```
Do we really need to use a library just to handle permissions?
@antailbaxt3r what do you think?

antailbaxt3r replied:
Yes, use EasyPermissions

devansh-299 commented on app/src/main/java/club/devsoc/scanf/view/activity/QRScanActivity.kt:
```diff
@@ -0,0 +1,75 @@
+            codeScanner.startPreview();
+        } else {
+            EasyPermissions.requestPermissions(this,
+                "Would you like to give this app the permissions to access Camera?",
```
Instead of hardcoding the message, please use a string resource

devansh-299 commented on app/src/main/java/club/devsoc/scanf/view/activity/QRScanActivity.kt:
```diff
@@ -0,0 +1,75 @@
+        super.onPause()
+    }
+
+    override fun onRequestPermissionsResult(requestCode: Int, permissions: Array<out String>, grantResults: IntArray) {
```
Avoid using more than 100 characters in a single line of code

devansh-299 commented on app/src/main/res/layout/activity_q_r_scan.xml:
```diff
@@ -0,0 +1,26 @@
+        app:flashButtonColor="@android:color/white"
+        app:flashButtonVisible="true"
+        app:frameColor="@android:color/white"
+        app:frameCornersSize="50dp"
```
Instead of using hardcoded values, use create/use resources. Same goes for other values too

ayush-sharma2601 replied:
Done, replaced hardcoded values with apt resources

devansh-299 replied:
> Done, replaced hardcoded values with apt resources

Resolve the merge conflict and squash the commits into one single commit. Also always follow snake casing while naming the resources. Have a look at this article - [link](https://medium.com/mindorks/android-resource-naming-convention-42e4e8026614)

ayush-sharma2601 committed:
Second Commit to QR Scan Activity. Changes:
- replaced hardcoded values in XML and Kotlin files with String/Dimen resources

ayush-sharma2601 committed:
Third Commit to QR Scan Activity. Changes:
- refactored the resources via corrected code style

ayush-sharma2601 committed:
Merge branch 'master' into master

antailbaxt3r commented:
Please solve the conflicts ASAP

ayush-sharma2601 committed:
Merge branch 'master' into scanfQRScanner
